testing.md

```diff
@@ -1,21 +1,137 @@
+
+##### Run script to start a local ADF instance
+```
+./resources/run_adf.sh start
+```
+#### Install mongoimport
+Download and install [MongoDB Command Line Database Tools](https://www.mongodb.com/try/download/database-tools)
+Add mongoimport.exe to the `PATH` and change permissions to make it executable.
+#### Load Sample Dataset
+```
+mongoimport.exe --uri="mongodb://$ADF_TEST_LOCAL_HOST:$MDB_TEST_LOCAL_PORT/supplies" \
+            --drop resources/integration_test/testdata/sales.json
+mongoimport.exe --uri="mongodb://$ADF_TEST_LOCAL_HOST:$MDB_TEST_LOCAL_PORT/integration_test" \
+            --drop resources/integration_test/testdata/complex_types.json
+```
+#### Generate Schema
+Generate the schema for the data that was loaded using the `mongo.exe` executable downloaded by the `run_adf.sh` script
+```
+MONGOSHELL=$(find ./local_adf/ | grep mongo.exe | head -1)
+chmod +x $MONGOSHELL
+$MONGOSHELL -u $ADF_TEST_LOCAL_USER --password $ADF_TEST_LOCAL_PWD --authenticationDatabase \
+            $ADF_TEST_LOCAL_AUTH_DB $ADF_TEST_LOCAL_HOST/admin \
+            --eval 'db.runCommand({sqlGenerateSchema: 1,
+            sampleNamespaces: ["integration_test.complex_types", "supplies.sales"], setSchemas: true})'
+```
+#### Clear Power BI Cache
+It is a good idea to clear the data cache and data source settings to ensure the connector is being correctly tested.
+##### Clear Data Cache
+* Navigate to `File` -> `Options and settings` -> `Options` -> `Data Load`
+* Under `Data Cache Management Options` click on `Clear Cache`
+
+##### Clear Saved Data Source Settings
+* Navigate to `File` -> `Options and settings` -> `Data source settings`
+* For `Data sources in current file` and `Global permissions` choose `Clear Permissions`->`Clear All Permissions`
+
+## Running Tests
+### Navigation Table
+Test that the expected tables are shown in the navigation table and that data is loaded in the expected format.
+* `Get Data` -> `More...` -> `Database` -> `MongoDB Atlas SQL (Beta)`
+* Enter MongoDB URI: `mongodb://localhost/?ssl=false`
+* Enter Database: `integration_test`
+* Click `OK`
+* In the Navigator, expand the `integration_test` and `supplies` databases to show the underlying collections
+* Choose the `complex_types` collection and verify that the expected preview loads
+* Click `Transform Data`
+* Update the query to the following to show the table schema:
+```
+= Table.Schema(integration_test_Database{[Name="complex_types",Kind="Table"]}[Data])
+```
+* Confirm the `TypeName` and `NativeTypeName` columns have the expected values
+  * The `TypeName` should be `Text.Type` for the complex types
+* Transform the `array` column to JSON
+* Expand the list to new rows
+* Click `Close and Apply`
+* In `Visualizations` choose `Clustered column chart`
+* In `Data` choose `double` and `integer`, verify the visualization is as expected
+
+### Query
```

Comment:
@terakilobyte What's the difference between this section and `Manual Testing (for Native Query)`?

Reply:
Nothing much except for what is being tested!